### ISSUE TITLE
isn't it supose to be "buffer2" instead of "buffer"

### ISSUE THREAD

**Issue by abdo-tahar** [ing-bank/zkrp#8]
https://github.com/ing-bank/zkrp/blob/97a3cddb56272ce874966f9ceb1e302718fd94c4/bulletproofs/util.go#L74

**Comment by mvdbos** [comment]
I agree it is confusing like this, but I am not sure that is correct. 
As you can see, the contents of `buffer` and `buffer2` are identical until [line 73](https://github.com/ing-bank/zkrp/blob/97a3cddb56272ce874966f9ceb1e302718fd94c4/bulletproofs/util.go#L73). There the result of the first hash is added. 

Did you test what happens if you change it?

**Comment by abdo-tahar** [comment]
sorry for the late respense , yes i have tested it and its wrong like that (result1 and result2 are the results of fiat-shamir heuristic for making two challenges ,the two challenges must not have the same value) so it should be buffer2

**Comment by emmorais** [comment]
yes, it was a mistake.
